src/v/raft/state_machine_manager.cc

```diff
@@ -240,8 +245,18 @@ ss::future<> state_machine_manager::stop() {
     _as.request_abort();
 
     auto gate_f = _gate.close();
-    co_await ss::coroutine::parallel_for_each(
-      _machines, [](auto p) { return p.second->stm->stop(); });
+    co_await ss::coroutine::parallel_for_each(_machines, [this](auto p) {
+        ssx::watchdog(
+          _stm_shutdown_timeout(), [ntp = _raft->ntp(), name = p.first] {
+              vlog(
+                raftlog.error,
+                "[{}] Timedout waiting for {} state machine to stop",
+                ntp,
+                name);
+          });
+        // stop the state machine
+        return p.second->stm->stop();
```

Comment:
does this work as expected? doesn't watchdog get destroyed immediately since we are returning a future?